src/app/MessageProvider.tsx

```diff
@@ -1,34 +1,47 @@
 import { shouldPolyfill } from '@formatjs/intl-pluralrules/should-polyfill';
-import React from 'react';
-import { createIntl, createIntlCache, RawIntlProvider } from 'react-intl';
+import React, { useState } from 'react';
+import { RawIntlProvider } from 'react-intl';
+import { useSelector } from 'react-redux';
+import { book as bookSelector } from './content/selectors';
 import { useServices } from './context/Services';
-import enMessages from './messages/en';
+import { pathname as pathSelector } from './navigation/selectors';
 
-// https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
-async function polyfill(locale: 'en') {
-  if (shouldPolyfill()) {
-    await import('@formatjs/intl-pluralrules/polyfill');
-  }
+// tslint:disable-next-line:variable-name
+const MessageProvider = (props: { children?: React.ReactNode }) => {
+  const book = useSelector(bookSelector);
+  const currentPath = useSelector(pathSelector);
+  const [polyfillLoaded, setPolyfillLoaded] = useState(false);
+
+  // https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
+  async function polyfill(locale: string | undefined) {
```

Comment:
this doesn't have to be defined in the function, you can `useEffect` with the locale as a dependency. something like:

```
useEffect(async () => {
  await polyfill(locale)
  setPolyfillLoaded(true);
}, [locale, polyfill])
```